modules/highlight/highlight_test.go

```diff
@@ -8,100 +8,116 @@ import (
+		})
+	}
+}
+
+func TestPlainText(t *testing.T) {
+
+	tests := []struct {
+		name string
+		code string
+		want []string
+	}{
+		{
+			name: "empty.py",
+			code: "",
+			want: []string{""},
 		},
 		{
-			name:     ".drone.yml - trailing space",
-			numLines: 13,
-			fileName: ".drone.yml",
-			code: strings.Replace(util.Dedent(`
-				kind: pipeline
-				name: default
-
-				steps:
-				- name: test
-					image: golang:1.13
-					environment:
-						GOPROXY: https://goproxy.cn
-					commands:
-					- go get -u
-					- go build -v
-					- go test -v -race -coverprofile=coverage.txt -covermode=atomic
-			`)+"\n", "name: default", "name: default  ", 1),
-			want: util.Dedent(`
-				<span class="nt">kind</span><span class="p">:</span><span class="w"> </span><span class="l">pipeline</span><span class="w">
-				</span><span class="w"></span><span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="l">default  </span><span class="w">
-				</span><span class="w">
+			name: "eol-no.py",
+			code: "a=1",
+			want: []string{`a=1`},
+		},
+		{
+			name: "eol-newline1.py",
+			code: "a=1\n",
+			want: []string{
+				`a=1&#10;`,
+			},
+		},
+		{
+			name: "eol-newline2.py",
+			code: "a=1\n\n",
+			want: []string{
+				`a=1&#10;`,
+				`&#10;`,
+			},
+		},
+		{
+			name: "empty-line.py",
+			code: strings.TrimSpace(`
+a=1
 
-				</span><span class="w"></span><span class="nt">steps</span><span class="p">:</span><span class="w">
-				</span><span class="w"></span>- <span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="l">test</span><span class="w">
-				</span><span class="w">	</span><span class="nt">image</span><span class="p">:</span><span class="w"> </span><span class="l">golang:1.13</span><span class="w">
-				</span><span class="w">	</span><span class="nt">environment</span><span class="p">:</span><span class="w">
-				</span><span class="w"></span><span class="w">		</span><span class="nt">GOPROXY</span><span class="p">:</span><span class="w"> </span><span class="l">https://goproxy.cn</span><span class="w">
-				</span><span class="w">	</span><span class="nt">commands</span><span class="p">:</span><span class="w">
-				</span><span class="w"></span><span class="w">	</span>- <span class="l">go get -u</span><span class="w">
-				</span><span class="w">	</span>- <span class="l">go build -v</span><span class="w">
-				</span><span class="w">	</span>- <span class="l">go test -v -race -coverprofile=coverage.txt -covermode=atomic</span><span class="w">
-				</span>
-				<span class="w">
-				</span>
+b=''
 			`),
+			want: []string{
+				`a=1&#10;`,
+				`&#10;`,
+				`b=&#39;&#39;`,
+			},
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got := strings.Join(File(tt.numLines, tt.fileName, "", []byte(tt.code)), "\n")
-			assert.Equal(t, tt.want, got)
-			assert.Equal(t, strings.Count(got, "<span"), strings.Count(got, "</span>"))
```

Comment:
Balanced tags test should not be removed.

Reply:
But the algorithm is stable and it won't generated unbalanced tags.

And since the test data is clear, you can easily see that all tags in the tests are balanced.

At the moment I do not think we need to count the tags again.

Reply:
Added back. 

But IMO it's not that useful. Even if the counting is correct, the tags can sill be unbalanced: `</span><span>`